dask_jobqueue/core.py

```diff
@@ -407,5 +409,11 @@ def _del_pending_jobs(self):
         return jobs
 
     def _job_id_from_submit_output(self, out):
-        raise NotImplementedError('_job_id_from_submit_output must be implemented when JobQueueCluster is '
-                                  'inherited. It should convert the stdout from submit_command to the job id')
+        regexp = r'(?P<job_id>{})'.format(self.job_id_regexp)
+        match = re.search(regexp, out)
```

Comment:
I think to cover all the cases that can error with a not very helpful error message, you need to do something like this:
```py
msg = 'Could not parse ...'

if match is None:
    raise ValueError(msg)

job_id = match.groupdict().get('job_id')

if job_id is None:
    raise ValueError(msg)
```

It would be great to have a test for error messages: you can use something like:
```py
with pytest.raises(ValueError, match=error_message_regexp):
    ...
```

Reply:
Done.

Reply:
The `match` keyword gets only a part of the expected error message. Not sure if this is enough. Looks similar to similar tests in `dask-ml`, though.